app/src/main/java/android/coding/interview/makeitawesome/fragment/PicturesFragment.java

```diff
@@ -25,13 +46,70 @@ public static Fragment newInstance() {
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        RecyclerView rv = (RecyclerView) inflater.inflate(R.layout.pictures_list_fragment, container, false);
+        /*
+        RecyclerView rv = (RecyclerView) inflater.inflate(R.layout.album_list_fragment, container, false);
         setupRecyclerView(rv);
+        */
+        rv = (ListView) inflater.inflate(R.layout.album_list_fragment, container, false);
+        DownloadFilesTask d = new DownloadFilesTask();
+        d.execute();
```

Comment:
For me, this is by far the worst part. He solves everything with an async task called during onCreateView, what is a quite bad idea. What about Android lifecycle, no clear layers... he just puts everything together. It probably works, but if the problem would have been slightly bigger it would probably grow into a big mess.